pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -31,9 +30,21 @@ def master_port(self) -> int:
         """ An open and configured port in the master node through which all processes communicate. """
 
     @abstractmethod
-    def world_size(self) -> Optional[int]:
+    def world_size(self) -> int:
         """ The number of processes across all devices and nodes. """
 
+    @abstractmethod
+    def set_world_size(self, size: int) -> None:
+        pass
+
+    @abstractmethod
+    def global_rank(self) -> int:
+        """ The rank (index) of the currently running process across all nodes and devices. """
```

Comment:
Shouldn't this be `pass` too?

Reply:
it doesn't make a difference. pass is only required when we have nothing under the function. here the docstring is already enough :)